great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -709,9 +709,12 @@ def _add_fluent_datasource(self, datasource: FluentDatasource) -> None:
                 f"Can not write the fluent datasource {datasource.name} because a datasource of that "
                 "name already exists in the data context."
             )
+        # store update here?
         self.datasources[datasource.name] = datasource
+        self._datasource_store.set(key=None, value=datasource)
 
     def _update_fluent_datasource(self, datasource: FluentDatasource) -> None:
+        # store update here?
         self.datasources[datasource.name] = datasource
```

Comment:
We don't seem to actually call this anywhere in our code right now.
The `context.sources.add_or_update_<DATASOURCE_TYPE>()` method always calls `_add_fluent_datasource`.